pennylane/tape/jacobian_tape.py

```diff
@@ -580,9 +587,15 @@ def jacobian(self, device, params=None, **options):
 
             if jac is None:
                 # update the tape's output dimension
-                self._output_dim = len(g)
-                # create the Jacobian matrix
-                jac = np.zeros((len(g), len(params)), dtype=float)
+                try:
+                    self._output_dim = len(g)
+                except TypeError:
+                    # if g has no len (e.g., because it is not a numpy.ndarray)
+                    # assume the dimension is 1
+                    self._output_dim = 1
+                # create the Jacobian matrix with appropriate dtype
+                dtype = float if isinstance(g, float) else np.object
+                jac = np.zeros((self._output_dim, len(params)), dtype=dtype)
```

Comment:
@cvjjm I noticed that [quite a few of the tests are failing](https://github.com/PennyLaneAI/pennylane/pull/1291/checks?check_run_id=2566860157), specifically the `np.allclose` calls within the tests. I'm not 100% sure why, but I think the issue might be the logic here?

Should it instead be tied to the existence of a `__len__` attribute?
```python
dtype = float if hasattr(g, "__len__") else np.object
```

Reply:
I wasn't sure what I need to put here to preserve the default behavior. I will look into this.